packages/devel/glibc/package.mk

```
@@ -123,12 +123,14 @@ post_makeinstall_target() {
   safe_remove ${INSTALL}/usr/lib/*.map
   safe_remove ${INSTALL}/var
 
-# add UTF-8 charmap for Generic (charmap is needed for installer)
-  if [ "${PROJECT}" = "Generic" ]; then
-    mkdir -p ${INSTALL}/usr/share/i18n/charmaps
-    cp -PR ${PKG_BUILD}/localedata/charmaps/UTF-8 ${INSTALL}/usr/share/i18n/charmaps
-    pigz --best --force ${INSTALL}/usr/share/i18n/charmaps/UTF-8
-  fi
+# add UTF-8 charmap
+  mkdir -p ${INSTALL}/usr/share/i18n/charmaps
+    cp -PR ${INSTALL}/.noinstall/charmaps/UTF-8.gz ${INSTALL}/usr/share/i18n/charmaps
+
+# compile locale definition en_GB.UTF-8
+  mkdir -p ${INSTALL}/usr/share/locpath
+    I18NPATH=${INSTALL}/.noinstall \
+      localedef -i en_GB -f UTF-8 ${INSTALL}/usr/share/locpath/en_GB.UTF-8
```

Distributions are using C.UTF-8 included to their base image because of much smaller size. sys.getfilesystemencoding() is changed to utf-8 too.

```suggestion
      localedef -i POSIX -f UTF-8 ${INSTALL}/usr/share/locpath/C.UTF-8 || test $? -le 1
```
The error code test is needed due to some warnings during compilation.

`export LANG=` has to be changed but second GitHub suggestion cannot be opened in one comment.

---

Thanks, will make the changes.

What error happen with calling localedef ? If it fails than this is serious error.

---

Use `LANG=C.UTF-8 `. (Was missing in my message).

Error code is set on warnings:
```
EXIT STATUS
       One of the following exit values can be returned by localedef:
       0      Command completed successfully.
       1      Warnings or errors occurred, output files were written.
       4      Errors encountered, no output created.
```

-c argument looks reasonable.